scripts/bluesky-0MQ-proxy

```diff
@@ -1,6 +1,78 @@
 #!/usr/bin/env python3
+import threading
 import argparse
+import logging
+import sys
 
+from bluesky.callbacks.zmq import RemoteDispatcher
+
+class Dispatcher_thread(threading.Thread):
+    def __init__(self, logfile, poison_pill):
+        super(Dispatcher_thread, self).__init__()
+        self._logfile = logfile
+        self._poison_pill = poison_pill
+        self._dispatcher = None
+
+    def set(self, poison_pill):
+        self._dispatcher.set_pill(poison_pill)
+
+    def run(self):
+        self._dispatcher = RemoteDispatcher(f'localhost:{out_port}', self._poison_pill)
+        print("self._logfile")
+        print(self._logfile)
+        if self._logfile:
+            print("inside log")
+            logger = logging.getLogger('logfile')
+            hdlr = logging.FileHandler(self._logfile)
+            formatter = logging.Formatter('%(asctime)s %(levelname)s %(message)s')
+            hdlr.setFormatter(formatter)
+            logger.addHandler(hdlr)
+            logger.setLevel(logging.INFO) #Only write to file with logger.info or worse
+            def log_writer(name, doc):
+                """logger's wrapper function
+                    This function will be use to fit .subscribe() method.
+                    It has two arguments as .subscribe expect. Inside, it
+                    call logger.info to write doc which is a dict as a str
+                    into logfile
+                """
+                logger.info(str(doc))
+                print("Log",name ,"doc to", self._logfile)
+            self._dispatcher.subscribe(log_writer) # Subscribe log writer
+        else:
+            self._dispatcher.subscribe(print)
+        self._dispatcher.start()
+            
+
+def start_dispatcher(logfile, poison_pill):
+    """The dispatcher function
+    Parameters
+    ----------
+    logfile : string
+        string come from user command. ex --logfile=temp.log
+        logfile will be "temp.log". logfile could be empty.
+    """
+    d = RemoteDispatcher(f'localhost:{out_port}')
```

Comment:
Should we make the machine configurable, rather having `localhost` hard-coded?
Also, one-letter vars is not considered a good practice, consider changing it to something more verbose.

Reply:
In general the user can point `RemoteDispatcher` at whatever they like. _This_ `RemoteDispatcher`'s whole job is to listen to the `Proxy` that we are about to start below (so, definitely on `localhost`) and generate log messages.